### PR TITLE
Update to the latest version of swiftpoet 

### PR DESCRIPTION
**Pull request by pickens117**
The latest version includes the escaped reserved type names changes.

### CODE REVIEW

**Review by JakeWharton — APPROVED** [review]

**Comment by oldergod** [comment]
Why the draft? Let's merge?

File: wire-library/wire-runtime-swift/src/test/proto/type.proto [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+syntax = "proto2";
+
+enum Type {
+  UNKNOWN = 0;
+  VALUE = 1;
+}
```
Comment (oldergod) [inline wire-library/wire-runtime-swift/src/test/proto/type.proto:21]: What's that for?

Comment (pickens117) [reply wire-library/wire-runtime-swift/src/test/proto/type.proto:21]: Prior to bumping swiftpoet to the latest version, the `:wire-runtime-swift:generateSwiftProtos` gradle task would  fail when attempting to generate the swift version of that proto. I left it in as a "pseudo" integration test. Happy to remove it if you think it's uanessacry or add a unit test specifically tested the generated swift file from the proto.

Comment (oldergod) [reply wire-library/wire-runtime-swift/src/test/proto/type.proto:21]: Then please add a comment to the proto file on the type explaining why it's there. And it's good!

Comment (pickens117) [reply wire-library/wire-runtime-swift/src/test/proto/type.proto:29]: Done!

**Pull request merged**